authority/ssh.go

```diff
@@ -198,6 +198,14 @@ func (a *Authority) SignSSH(ctx context.Context, key ssh.PublicKey, opts provisi
 				errs.WithKeyVal("signOptions", signOpts),
 			)
 		}
+		// explicitly check for unmarshaling errors, which are most probably caused by JSON template syntax errors
+		if strings.HasPrefix(err.Error(), "error unmarshaling certificate") {
+			msg := strings.TrimSpace(strings.TrimPrefix(err.Error(), "error unmarshaling certificate:"))
+			return nil, errs.ApplyOptions(
+				errs.InternalServer("authority.Sign: failed to apply certificate template: %s", msg),
+				errs.WithKeyVal("signOptions", signOpts),
+			)
+		}
```

Comment:
I have an unfinished PR to write better internal errors, where we send users nicer errors and write in the logs more details if available. Without my PR this internal server error will send a very opaque error to the CLI.

We can do a few changes to follow this:
* I'm removing is those `authority.Sign` headers, so remove that.
* Although in my PR is different (easier), in this code we already have a way to send a message to the CLI using the options `errs.WithMessage()` or something similar. We can send the users something saying that there was an error processing the certificate template without specifying more details.
* `errs.ApplyOptions` is something that I added on my PR to add support for this k/v details, I'm not sure if that works with "old" error types like that one.

I think we can write the return as:

```go
return nil, errs.InternalServerErr(err, 
    errs.WithKeyVal("signOptions", signOpts),
    errs.WithMessage("error processing certificate template (or something similar)"),
)
```

That will write the original error to the logs and send the user the message that we want. 

I don't think is necessary to trim the message that we write in the logs, but if you think you need you can use `errors.Cause` (in pkg/errors, returns the wrapped error). It's even possible with `errors.As` to get the `json.SyntaxError`, but there might be other types of errors, so it's not that easy.